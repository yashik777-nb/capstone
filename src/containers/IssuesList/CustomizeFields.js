import React from "react";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core/styles";
import { blue } from "@material-ui/core/colors";
import FormGroup from "@material-ui/core/FormGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import * as actionCreators from "../../store/actions/actionsIndex";

export const BlueCheckBox = withStyles({
  root: {
    color: blue[400],
    "&$checked": {
      color: blue[600],
    },
  },
  checked: {},
})((props) => <Checkbox color="default" {...props} />);

class CustomizeFields extends React.Component {
  render() {
    return (
      <div>
        <h6>Customize Fields</h6>
        <FormGroup row>
          <FormControlLabel
            control={
              <BlueCheckBox
                checked={this.props.severity}
                onChange={(event) => this.props.severityHandler()}
                name="severity"
              />
            }
            label="Severity"
          />
          <FormControlLabel
            control={
              <BlueCheckBox
                checked={this.props.status}
                onChange={(event) => this.props.statusHandler()}
                name="status"
              />
            }
            label="Status"
          />
          <FormControlLabel
            control={
              <BlueCheckBox
                checked={this.props.createdDate}
                onChange={(event) => this.props.createdDateHandler()}
                name="creaedDate"
              />
            }
            label="Created Date"
          />
          <FormControlLabel
            control={
              <BlueCheckBox
                checked={this.props.resolvedDate}
                onChange={(event) => this.props.resolvedDateHandler()}
                name="resolvedDate"
              />
            }
            label="Resolved Date"
          />
        </FormGroup>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    ...state.customize,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    severityHandler: () => dispatch(actionCreators.severityHandler()),
    statusHandler: () => dispatch(actionCreators.statusHandler()),
    createdDateHandler: () => dispatch(actionCreators.createdDateHandler()),
    resolvedDateHandler: () => dispatch(actionCreators.resolvedDateHandler()),
  };
};

export default withStyles({})(
  connect(mapStateToProps, mapDispatchToProps)(CustomizeFields)
);
