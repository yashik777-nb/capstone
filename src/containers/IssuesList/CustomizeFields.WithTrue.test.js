import React from "react";
import { Provider } from "react-redux";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import configureStore from "redux-mock-store";

import CustomizeFields, { BlueCheckBox } from "./CustomizeFields";

configure({ adapter: new Adapter() });

const store = {};

const props = {
  severity: true,
  status: true,
  createdDate: true,
  resolvedDate: true,
  severityHandler: () => {},
  statusHandler: () => {},
  createdDateHandler: () => {},
  resolvedDateHandler: () => {},
};

const mockStore = configureStore();

describe("Navigation Bar Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <CustomizeFields {...props} />
      </Provider>
    );
  });

  it("Should render four BlueCheckbox components", () => {
    expect(wrapper.find(BlueCheckBox).length).toEqual(4);
  });

  it("Should render the severity checkbox as checked", () => {
    expect(wrapper.find(BlueCheckBox).at(0).props().checked).toEqual(true);
  });

  it("Should render the status checkbox as checked", () => {
    expect(wrapper.find(BlueCheckBox).at(1).props().checked).toEqual(true);
  });

  it("Should render the created date checkbox as checked", () => {
    expect(wrapper.find(BlueCheckBox).at(2).props().checked).toEqual(true);
  });

  it("Should render the resolved date checkbox as checked", () => {
    expect(wrapper.find(BlueCheckBox).at(3).props().checked).toEqual(true);
  });

  test("Should Render Customize Fields Component With Props as true", () => {
    expect(wrapper).toMatchSnapshot();
  });

});
