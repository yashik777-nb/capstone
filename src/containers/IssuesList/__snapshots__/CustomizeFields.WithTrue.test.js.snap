// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Tests Should Render Customize Fields Component With Props as true 1`] = `ReactWrapper {}`;
